posthog/api/action.py

```diff
@@ -213,10 +217,15 @@ def _aggregate_by_day(self, action: Action, filters: Dict[Any, Any], request: re
             dates_filled = self._group_events_to_date(date_from=date_from, date_to=date_to, aggregates=aggregates)
             append = self._append_data(append, dates_filled)
         if request.GET.get('breakdown'):
-            append['breakdown'] = self._breakdown(aggregates, breakdown_by=request.GET['breakdown'])
+            append = self._breakdown(append, action, filters,breakdown_by=request.GET['breakdown'])
```

Comment:
this is funky to begin with because if there's an applied breakdown the frontend doesn't actually need the other data to display the breakdown chart and because breakdown isn't actually part of the aggregate by day idea. I didn't refactor this area for now because I didn't want to break frontend atm. will revisit

Reply:
Yep you're right, we could have an if statement somewhere to toggle between.